### DISCUSSION TITLE
Windows blank screen?

### DISCUSSION THREAD

**Discussion by dgrahn** [Textualize/textual#335] [Q&A]
I'm trying to run this code through anaconda prompt on windows. All I see is a blank screen. 😢 I installed from pip, version 0.1.17.

```
from textual.app import App
from textual.widgets import Placeholder


class SimpleApp(App):

    async def on_mount(self) -> None:
        await self.view.dock(Placeholder(), edge="left", size=40)
        await self.view.dock(Placeholder(), Placeholder(), edge="top")


SimpleApp.run(log="textual.log")
```

**Comment by LuminairPrime** [comment]
Just downloaded Textual now, and I noticed that all the examples work when manually run by typing the command in the command line, but not when double clicking the python file...

**Reply by USLTD** [reply]
How many **Python** instances have you installed?

**Reply by LuminairPrime** [reply]
I personally have Python 3.10.1 installed, and I only see Python310 in the path variable...

**Reply by USLTD** [reply]
So you have only **ONE** <kbd>**Python**</kbd> version/instance installed on entire computer right?
Can you upload your terminal and OS details?
Also it would be preferred to upload output from bellow commands
```
python -m rich.diagnose
pip freeze | grep rich # only if you have grep installed
```

**Reply by LuminairPrime** [reply]
Windows 10 for me. FYI there is another similar report: https://github.com/Textualize/textual/discussions/321

The rich diagnose works when manually entered in cmd.exe, but so do the examples when manually entered. Also, other GUI library examples I tried double-clicking DO work, like asciimatics.

So, there's something going wrong between the Python application window that Windows launches when double-clicking a .py file, and Textualize. Maybe this is related, or unrelated, to the OP's Anaconda problem....

**Reply by USLTD** [reply]
@LuminairPrime , what is default app set for `.py` extension on your computer?

**Reply by LuminairPrime** [reply]
Just the default. It has that blue and yellow Python logo at the top left, and the path is: users, username, appdata, local, programs, python, launcher, py.exe

**Comment by dgrahn** [comment]
Could you clarify the question?

I have Anaconda installed. No stand-alone versions of Python (just those
integrated into the environments).

**Reply by USLTD** [reply]
*What I meant is how many different Python versions do you have installed*.
Also, can you try to install **Textual** from this **GitHub** repo? Using [this answer](https://stackoverflow.com/a/50141879/10944855)?
If it doesn't work then please use following commands to get diagnose and then upload result (preferred in TXT or `code` format, but screenshot should also do the job).

**Reply by USLTD** [reply]
`python -m rich.diagnose`

**Reply by USLTD** [reply]
If you have **grep** then please run following command too
`pip freeze | grep rich`

**Reply by dgrahn** [reply]
```
┌───────────────────────── <class 'rich.console.Console'> ─────────────────────────┐
│ A high level console interface.                                                  │
│                                                                                  │
│ ┌──────────────────────────────────────────────────────────────────────────────┐ │
│ │ <console width=160 ColorSystem.WINDOWS>                                      │ │
│ └──────────────────────────────────────────────────────────────────────────────┘ │
│                                                                                  │
│     color_system = 'windows'                                                     │
│         encoding = 'utf-8'                                                       │
│             file = <_io.TextIOWrapper name='<stdout>' mode='w' encoding='utf-8'> │
│           height = 40                                                            │
│    is_alt_screen = False                                                         │
│ is_dumb_terminal = False                                                         │
│   is_interactive = True                                                          │
│       is_jupyter = False                                                         │
│      is_terminal = True                                                          │
│   legacy_windows = True                                                          │
│         no_color = False                                                         │
│          options = ConsoleOptions(                                               │
│                        size=ConsoleDimensions(width=160, height=40),             │
│                        legacy_windows=True,                                      │
│                        min_width=1,                                              │
│                        max_width=160,                                            │
│                        is_terminal=True,                                         │
│                        encoding='utf-8',                                         │
│                        max_height=40,                                            │
│                        justify=None,                                             │
│                        overflow=None,                                            │
│                        no_wrap=False,                                            │
│                        highlight=None,                                           │
│                        markup=None,                                              │
│                        height=None                                               │
│                    )                                                             │
│            quiet = False                                                         │
│           record = False                                                         │
│         safe_box = True                                                          │
│             size = ConsoleDimensions(width=160, height=40)                       │
│        soft_wrap = False                                                         │
│           stderr = False                                                         │
│            style = None                                                          │
│         tab_size = 8                                                             │
│            width = 160                                                           │
└──────────────────────────────────────────────────────────────────────────────────┘
┌─── <class 'rich._windows.WindowsConsoleFeatures'> ────┐
│ Windows features available.                           │
│                                                       │
│ ┌───────────────────────────────────────────────────┐ │
│ │ WindowsConsoleFeatures(vt=False, truecolor=False) │ │
│ └───────────────────────────────────────────────────┘ │
│                                                       │
│ truecolor = False                                     │
│        vt = False                                     │
└───────────────────────────────────────────────────────┘
┌────── Environment Variables ───────┐
│ {                                  │
│     'TERM': None,                  │
│     'COLORTERM': None,             │
│     'CLICOLOR': None,              │
│     'NO_COLOR': None,              │
│     'TERM_PROGRAM': None,          │
│     'COLUMNS': None,               │
│     'LINES': None,                 │
│     'JPY_PARENT_PID': None,        │
│     'VSCODE_VERBOSE_LOGGING': None │
│ }                                  │
└────────────────────────────────────┘
platform="Windows"
```

Obviously grep won't run on Windows. But I created a new virtual env for this, so there's not much installed.

```
python==3.7.11
commonmark==0.9.1
Pygments==2.11.2
rich==12.0.1
textual @ file:///C:/dev/temp/textual
typing-extensions==3.10.0.2
```

**Reply by USLTD** [reply]
I don't see anything wrong... What could it be...? 🤔
I even tested it on my **Windows** device (only not on **Anaconda** since I don't have one) but it worked without errors, except there was no color for unknown reasons.

Did you install **Textual** from **git** or **pip**?

**Reply by yehnan** [reply]
I have the same issue, that is, textual on Windows 10 only shows blank screen.

I use the following commands to install textual:
conda create -n python=3.7
conda activate py37
conda install git pip
pip install git+git://github.com/Textualize/textual@main

But
python -m textual.app
only shows blank screen.

rich installed version is: 12.0.1
textual installed version is 0.1.17


python -m rich.diagnose
┌───────────────────────── <class 'rich.console.Console'> ─────────────────────────┐
│ A high level console interface.                                                  │
│                                                                                  │
│ ┌──────────────────────────────────────────────────────────────────────────────┐ │
│ │ <console width=94 ColorSystem.WINDOWS>                                       │ │
│ └──────────────────────────────────────────────────────────────────────────────┘ │
│                                                                                  │
│     color_system = 'windows'                                                     │
│         encoding = 'utf-8'                                                       │
│             file = <_io.TextIOWrapper name='<stdout>' mode='w' encoding='utf-8'> │
│           height = 30                                                            │
│    is_alt_screen = False                                                         │
│ is_dumb_terminal = False                                                         │
│   is_interactive = True                                                          │
│       is_jupyter = False                                                         │
│      is_terminal = True                                                          │
│   legacy_windows = True                                                          │
│         no_color = False                                                         │
│          options = ConsoleOptions(                                               │
│                        size=ConsoleDimensions(width=94, height=30),              │
│                        legacy_windows=True,                                      │
│                        min_width=1,                                              │
│                        max_width=94,                                             │
│                        is_terminal=True,                                         │
│                        encoding='utf-8',                                         │
│                        max_height=30,                                            │
│                        justify=None,                                             │
│                        overflow=None,                                            │
│                        no_wrap=False,                                            │
│                        highlight=None,                                           │
│                        markup=None,                                              │
│                        height=None                                               │
│                    )                                                             │
│            quiet = False                                                         │
│           record = False                                                         │
│         safe_box = True                                                          │
│             size = ConsoleDimensions(width=94, height=30)                        │
│        soft_wrap = False                                                         │
│           stderr = False                                                         │
│            style = None                                                          │
│         tab_size = 8                                                             │
│            width = 94                                                            │
└──────────────────────────────────────────────────────────────────────────────────┘
┌─── <class 'rich._windows.WindowsConsoleFeatures'> ────┐
│ Windows features available.                           │
│                                                       │
│ ┌───────────────────────────────────────────────────┐ │
│ │ WindowsConsoleFeatures(vt=False, truecolor=False) │ │
│ └───────────────────────────────────────────────────┘ │
│                                                       │
│ truecolor = False                                     │
│        vt = False                                     │
└───────────────────────────────────────────────────────┘
┌────── Environment Variables ───────┐
│ {                                  │
│     'TERM': None,                  │
│     'COLORTERM': None,             │
│     'CLICOLOR': None,              │
│     'NO_COLOR': None,              │
│     'TERM_PROGRAM': None,          │
│     'COLUMNS': None,               │
│     'LINES': None,                 │
│     'JPY_PARENT_PID': None,        │
│     'VSCODE_VERBOSE_LOGGING': None │
│ }                                  │
└────────────────────────────────────┘
platform="Windows"

**Reply by dgrahn** [reply]
@UltraStudioLTD I tried from both pip and git. Are you using cmd or powershell?

**Reply by USLTD** [reply]
I am using CMD and WT (Windows Terminal. Both preview and stable). I have even tried the PuTTY and MingW terminal.

**Reply by USLTD** [reply]
Can you try in non-virtual environment. Outside of conda?

**Reply by USLTD** [reply]
Although I should also mention that I don't use **Conda**

**Comment by BoniLindsley** [comment]
Adding to the information available, this is a [repro video](https://user-images.githubusercontent.com/14114757/174439389-a0fd674b-f042-4571-8660-ee10ed2fbb6a.mp4). It shows the following

```cmd
py -m pip install --user --upgrade --compile textual
:: rich 12.4.1
:: textual 0.1.18

py -m rich.diganose
py -m rich
py -m textual.app
```

After launching `textual.app`,

- Resizing the console refreshed rendering, with artifacts.
- Mouse button pressed on `Q | Quit` exits app. Keyboard `q` also works.
- Held Enter to move prompt. Shows scroll buffer in new lines.
  - Mostly mitigated by disabling scroll buffer as in <https://superuser.com/a/1108493>. Sometimes the last line from app repeats without colours.

This is on Windows 10 Version 20H2, using Python 3.10.2 from <https://www.python.org/downloads/>.

Potentially related: <https://github.com/Textualize/textual/issues/305>.

Edit: Thank you to @DavinderJolly for the following:

- Textual works as is on Alacritty and Windows Terminal.
- Mostly works on PowerShell 7. May not clear screen after shutdown.
- Blank screen as illustrated for Windows Console and Windows PowerShell. However, calling `os.system("")` before creating `App` helps. Somehow.
